src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/ServiceHubLifetimeManagerFactory.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal class ServiceHubLifetimeManagerFactory
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ServiceManagerContext _context;
+        public ServiceHubLifetimeManagerFactory(IServiceProvider sp, IOptions<ServiceManagerContext> context)
+        {
+            _serviceProvider = sp;
+            _context = context.Value;
+        }
+        public async Task<IServiceHubLifetimeManager> Create(string hubName, CancellationToken cancellationToken, ILoggerFactory loggerFactoryPerHub=null)
+        {
+            switch (_context.ServiceTransportType)
+            {
+                case ServiceTransportType.Persistent:
+                    {
+                        var containerFactory = _serviceProvider.GetRequiredService<MultiEndpointConnectionContainerFactory>();
+                        var container = containerFactory.Create(hubName,loggerFactoryPerHub);
+                        var connectionManager = new ServiceConnectionManager<Hub>();
+                        connectionManager.SetServiceConnection(container);
+                        _ = connectionManager.StartAsync();
```

Comment:
This class is a factory for creating ServiceHubLifetimeManager, but it also start connection manager here. start should be called in other place

Reply:
Strictly speaking, `CreateServiceHubContext()` is a method for creating `ServiceHubContext`, but we start connection manager inside it, and this is unavoidable. Given this, I don't think starting connection manager in another factory is unacceptable?

Reply:
How about moving the preparison for connection manager into WebSocketsHubLifetimeManager? So that WebSocketsHubLifetimeManager is no knowledge of hub name, and is also able to put into service collection, and make it transient? And set the connection container and start the container in ServiceHubContextFactory.CreateAsync like the old version? @vicancy 

```
                        var containerFactory = _serviceProvider.GetRequiredService<MultiEndpointConnectionContainerFactory>();
                        var container = containerFactory.Create(hubName,loggerFactoryPerHub);
                        var connectionManager = new ServiceConnectionManager<Hub>();
                        connectionManager.SetServiceConnection(container);
```

Reply:
If I understand correctly, `WebSocketsHubLifetimeManager` is transient requires that `IServiceHubContext` is transient too?

Reply:
you plan to get a new one for each request?

Reply:
The key is that `HubContext<>` is an internal class, and we can't add it as transient.